inception-kb/src/test/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseServiceTest.java

```diff
@@ -0,0 +1,247 @@
+
+import de.tudarmstadt.ukp.clarin.webanno.model.Project;
+import de.tudarmstadt.ukp.inception.kb.graph.KBHandle;
+import de.tudarmstadt.ukp.inception.kb.model.KnowledgeBase;
+import de.tudarmstadt.ukp.inception.kb.reification.Reification;
+import de.tudarmstadt.ukp.inception.kb.util.TestFixtures;
+import de.tudarmstadt.ukp.inception.kb.yaml.KnowledgeBaseMapping;
+import de.tudarmstadt.ukp.inception.kb.yaml.KnowledgeBaseProfile;
+
+
+@RunWith(Parameterized.class)
+@SpringBootTest(classes = SpringConfig.class)
+@Transactional
+@DataJpaTest
+public class KnowledgeBaseServiceTest
+{
+    private static final String PROJECT_NAME = "Test project";
+    private String KB_NAME = "TestKB";
+
+    private KnowledgeBaseServiceImpl sut;
+    private Project project;
+    private KnowledgeBase kb;
+    private TestFixtures testFixtures;
+    private Reification reification;
+    
+    private final Logger log = LoggerFactory.getLogger(getClass());
+
+    
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @Autowired
+    private TestEntityManager testEntityManager;
+
+    @ClassRule
+    public static final SpringClassRule SPRING_CLASS_RULE = new SpringClassRule();
+
+    @Rule
+    public final SpringMethodRule springMethodRule = new SpringMethodRule();
+    
+    @BeforeClass
+    public static void setUpOnce() {
+        System.setProperty("org.eclipse.rdf4j.repository.debug", "true");
+    }
+    
+    public KnowledgeBaseServiceTest(Reification aReification) {
+        reification = aReification;
+    }
+
+    @Parameterized.Parameters(name = "Reification = {0}")
+    public static Collection<Object[]> data()
+    {
+        return Arrays.stream(Reification.values()).map(r -> new Object[] { r })
+            .collect(Collectors.toList());
+    }
+    
+    @Before
+    public void setUp() throws Exception {
+        EntityManager entityManager = testEntityManager.getEntityManager();
+        testFixtures = new TestFixtures(testEntityManager);
+        sut = new KnowledgeBaseServiceImpl(temporaryFolder.getRoot(), entityManager);
+        project = testFixtures.createProject(PROJECT_NAME);
+    }
+
+    @After
+    public void tearDown() throws Exception
+    {
+        testEntityManager.clear();
+        sut.destroy();
+    }
+    
+    @Test
+    public void testListPropertiesKnowledgeBaseIRI_WithLocalOWL() throws Exception
+    {
+        kb = buildKnowledgeBase(project, KB_NAME.concat("-" + "OWL"));
+        kb.setType(RepositoryType.LOCAL);
+        sut.registerKnowledgeBase(kb, sut.getNativeConfig());
+        setSchema(kb, OWL.CLASS, RDFS.SUBCLASSOF, RDF.TYPE, RDFS.COMMENT, RDFS.LABEL, RDF.PROPERTY);
+        importKnowledgeBase("data/wine-ontology.rdf");
+        List<KBHandle> propertiesKBHandle = sut.listProperties(kb,  RDF.PROPERTY, true, true);
+        log.debug("\n \n Size of List Properties OWL::::::::" + propertiesKBHandle.size() + "\n \n");
```

Comment:
Delete this pls

Reply:
@Rentier Test class in this branch is an older version.